writeengine/wrapper/writeengine.cpp

```diff
@@ -1690,34 +1929,70 @@ int WriteEngineWrapper::insertColumnRecs(const TxnID& txnid,
         int         curFbo = 0, curBio, lastFbo = -1;
 
         if (isFirstBatchPm && (totalRow == rowsLeft))
-        {}
+        {
+            idbassert(0 && "XXX: We insert rows only into second split LBID. We must add code to set extent's range."); // XXX
+        }
         else
         {
+            int firstHalfCount = totalRow - rowsLeft;
             for (unsigned i = 0; i < colStructList.size(); i++)
             {
                 colOp = m_colOp[op(colStructList[i].fCompressionType)];
                 width = colStructList[i].colWidth;
-                successFlag = colOp->calculateRowId(lastRid, BYTE_PER_BLOCK / width, width, curFbo, curBio);
+                if (firstHalfCount)
+		{
+                    BRM::CPInfo* cpInfoP = getCPInfoToUpdateForUpdatableType(colStructList[i], &maxMins[i].fSplitMaxMinInfo[0]);
+                    RID thisRid = rowsLeft ? lastRid : lastRidNew;
+                    successFlag = colOp->calculateRowId(thisRid, BYTE_PER_BLOCK / width, width, curFbo, curBio);
 
-                if (successFlag)
-                {
-                    if (curFbo != lastFbo)
+                    if (successFlag)
                     {
-                        RETURN_ON_ERROR(AddLBIDtoList(txnid,
-                                                      colStructList[i],
-                                                      curFbo,
-						      dummyCPInfo));
+                        if (curFbo != lastFbo)
+                        {
+                            RETURN_ON_ERROR(AddLBIDtoList(txnid,
+                                                          colStructList[i],
+                                                          curFbo, cpInfoP));
+                        }
                     }
-                }
+                    maxMins[i].fSplitMaxMinInfoPtrs[0] = cpInfoP;
+		}
+                if (rowsLeft)
+		{
+                    BRM::CPInfo* cpInfoP = getCPInfoToUpdateForUpdatableType(colStructList[i], &maxMins[i].fSplitMaxMinInfo[1]);
+		    if (cpInfoP)
+		    {
+                        RETURN_ON_ERROR(GetLBIDRange(newExtentsStartingLbids[i], colStructList[i], *cpInfoP));
+		    }
+
+                    maxMins[i].fSplitMaxMinInfoPtrs[1] = cpInfoP;
+		}
             }
         }
 
-        markTxnExtentsAsInvalid(txnid);
+        markTxnExtentsAsInvalid(txnid); // moved here as it can be source of bug.
```

Comment:
I didn't get the comment meaning b/c the placement of t he statement looks the same for me. Could you ellaborate on this?